Feat: reintegrate traffic storage

sfendrich opened the pull request:
The traffic storage has not been carried over to ORS 7. This commit reintegrates the reading and storing of traffic data. The reading of traffic data is disabled in the config and like-wise the corresponding API test because we currently have no freely publishable test data.

### Pull Request Checklist
<!--- Please make sure you have completed the following items BEFORE submitting a pull request (put an x in each box
when you have checked you have done them): -->
- [x] 1. I have [**rebased**][rebase] the latest version of the master branch into my feature branch and all conflicts
         have been resolved.
- [ ] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the
         [Unreleased] heading.
- [ ] 3. I have documented my code using JDocs tags.
- [x] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [ ] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [x] 6. I have created API tests for any new functionality exposed to the API.
- [ ] 7. If changes/additions are made to the ors-config.json file, I have added these to the [ors config documentation][config]
         along with a short description of what it is for, and documented this in the Pull Request (below).
- [x] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [x] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [ ] 10. For new features or changes involving building of graphs, I have tested on a larger dataset
          (at least Germany), and the graphs build without problems (i.e. no out-of-memory errors).
- [ ] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the
          importer etc.), I have generated longer distance routes for the affected profiles with different options
          (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end
          points generated from the current live ORS.
          If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [ ] 12. I have written in the Pull Request information about the changes made including their intended usage
          and why the change was needed.
- [ ] 13. For changes touching the API documentation, I have tested that the API playground [renders correctly][api].

Relates to [#1343](https://github.com/GIScience/openrouteservice/issues/1343)

### Information about the changes
- Key functionality added: Reintegration of traffic storages
- Reason for change: enable traffic dependent routing

Commit:
Feat: reintegrate traffic storage

The traffic storage has not been carried over to ORS 7. This commit
reintegrates the reading and storing of traffic data. The reading of
traffic data is disabled in the config and like-wise the corresponding
API test because we currently have no freely publishable test data.

aoles commented on openrouteservice/src/test/java/org/heigit/ors/apitests/isochrones/ResultTest.java:
```diff
@@ -478,4 +479,23 @@ void testCompleteMetadata() {
+
+    @Disabled
+    @Test
+    void testAcceptsTimeParameter() {
```
AFAIU the purpose of this test is to assess whether providing the time parameter in the case traffic data is enabled yields a valid response, and does not care about the actual content of the response. As such, it might be better harbored under `ParamsTest`, what do you think?

sfendrich replied:
I don't know. Looking through existing tests, the distinction between Params and Results seems arbitrary. I don't mind whether this test is here or there. I can put it where you wish.

aoles replied:
Thanks, maybe @jlink has some thoughts on this?

jlink replied:
Maybe there exists a better criterion than Param/Result for chunking the tests. What I do know is that many of the ResultTest classes have way too many tests to find my way around.

aoles replied:
Not sure what the best strategy is. As long as the test names are clear and descriptive enough I don't mind having the tests aggregated in classes grouped by endpoints. For now I would suggest we try to stick to what is already there and try to be consequent when introducing new stuff.  FWIU the distinction between `Param/Result` tests is that the former ones validate whether certain parameters yield meaningful response while the latter one actually deals with the correctness of the results. But of course, there is no clear distinction, and each one of use might have a slightly different perception where the borderline is. From my point of view it makes sense to move the test unless someone objects.

sfendrich replied:
O.K. I have moved the test into ParamsTests and created an issue on the cleanup project board.

aoles commented on openrouteservice/src/test/resources/ors-config.json:
```diff
@@ -192,6 +192,15 @@
+                },
+                "HereTraffic": {
+                  "enabled": false,
+                  "streets": "src/main/files/here_traffic/germany_traffic/Streets.shp",
```
I believe files which are not being included in the repo shouldn't be referenced in the config, unless I'm missing sth here?

sfendrich replied:
This is one of the missing mock files that should be added.

aoles commented on openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/ORSGraphHopper.java:
```diff
@@ -401,11 +404,29 @@ private LineString constructFreeHandRouteGeometry(GHRequest request) {
 	 */
 	@Override
 	protected void postProcessingHook(boolean closeEarly) {
+		matchTraffic(); // TODO: Must this be moved into GraphHopper like it was originally?
```
Re the TODO:  depending on whether traffic data is to be used with TD ALT, it might actually be necessary to have the data available before creating the landmarks. This would be in particular true when traffic speed could override the default edge-speed by setting it to a higher value. In such scenario the maximum speed value needs to be considered for calculating LMs in order to ensure the correctness of the approximator.

sfendrich replied:
So, is there anything to do for now?

aoles replied:
According to the implementation of `TrafficSpeedCalculator` traffic speed can actually override the edge speed by setting it to a higher value in [here](https://github.com/GIScience/openrouteservice/blob/master/openrouteservice/src/main/java/org/heigit/ors/routing/traffic/TrafficSpeedCalculator.java#L53-L66). Therefore, it is important that the landmarks are build only after traffic data has been loaded such that they can use use the maximum possible speed taking into account free flow traffic as in [here](https://github.com/GIScience/openrouteservice/blob/master/openrouteservice/src/main/java/org/heigit/ors/routing/traffic/TrafficSpeedCalculator.java#L44-L47).

sfendrich replied:
Moved matchTraffic into graphhopper

aoles reviewed (changes requested):
Thanks, please consider my inline comments. Cheers!

Commit:
Refactor(test): move testAcceptsTimeParameter to ParamsTests

See https://github.com/GIScience/openrouteservice/pull/1376

rabidllama committed:
Added sample traffic dataset for use in testing

Commit:
Fix(test): enable testAcceptsTimeParameter

aoles commented on openrouteservice/src/test/java/org/heigit/ors/apitests/isochrones/ParamsTest.java:
```diff
@@ -18,6 +18,7 @@
 import org.heigit.ors.apitests.common.VersionAnnotation;
 import org.json.JSONArray;
 import org.json.JSONObject;
+import org.junit.jupiter.api.Disabled;
```
Seems to be unused.

sfendrich replied:
Oh, sure, the test is enabled now. Removed.

Commit:
Fix: move traffic matching before graph preparations

Map matching for traffic needs to happen before the preparations of
speedup techniques in order to use them with traffic data.

Commit:
Refactor: remove unused import

aoles commented:
Thanks @sfendrich 🎉
